### 1. First Let's Import The Data Again

In [1]:
import pandas as pd
import numpy as np

In [2]:
data_preprocessed = pd.read_csv('df_EDA.csv')
data_preprocessed.head()

age  sex  cp  trtbps  chol  fbs  restecg  thalachh  exng  oldpeak  slp  \
0   63    1   3     145   233    1        0       150     0      2.3    0   
1   37    1   2     130   250    0        1       187     0      3.5    0   
2   41    0   1     130   204    0        0       172     0      1.4    2   
3   56    1   1     120   236    0        1       178     0      0.8    2   
4   57    0   0     120   354    0        1       163     1      0.6    2   

   caa  thall  output  
0    0      1       1  
1    0      2       1  
2    0      2       1  
3    0      2       1  
4    0      2       1

### 2. Train Test Split

In [3]:
X=data_preprocessed.drop(["output"],axis=1)
y=data_preprocessed["output"]

In [4]:
from sklearn.model_selection import train_test_split

X_train,X_test,y_train,y_test = train_test_split(X, y, test_size = 0.2, random_state = 40)

### 3. Modeling And Model Validation

In [5]:
from sklearn.linear_model import LogisticRegression
from sklearn.model_selection import KFold
from sklearn.neighbors import KNeighborsClassifier
from sklearn.svm import SVC
from sklearn.tree import DecisionTreeClassifier
from sklearn.ensemble import RandomForestClassifier
from sklearn.naive_bayes import GaussianNB
from sklearn.model_selection import cross_val_score

### 3.1 LogisticRegression

In [6]:
reg = LogisticRegression(max_iter = 2000)
scoring = 'accuracy'
score = cross_val_score(reg, X, y, n_jobs=1, scoring=scoring)
print(score)
round(np.mean(score)*100,2)

[0.81967213 0.8852459  0.85       0.88333333 0.75      ]


83.77

### 3.2 Cross Validation (K-fold)

In [7]:
k_fold = KFold(n_splits=10, shuffle=True, random_state=0)

### 3.2.3 kNN Classifier

In [8]:
clf = KNeighborsClassifier(n_neighbors = 13)
scoring = 'accuracy'
score = cross_val_score(clf, X, y, cv=k_fold, n_jobs=1, scoring=scoring)
print(score)
round(np.mean(score)*100, 2)

[0.70967742 0.64516129 0.66666667 0.56666667 0.73333333 0.76666667
 0.63333333 0.56666667 0.63333333 0.56666667]


64.88

### 3.2.4 Decision Tree Classifier

In [9]:
DTC = DecisionTreeClassifier()
scoring = 'accuracy'
score = cross_val_score(DTC, X, y, cv=k_fold, n_jobs=1, scoring=scoring)
print(score)
round(np.mean(score)*100, 2)

[0.74193548 0.74193548 0.7        0.83333333 0.7        0.83333333
 0.7        0.83333333 0.76666667 0.7       ]


75.51

### 3.2.5 Random Forest Classifier

In [10]:
RFC = RandomForestClassifier(n_estimators=13)
scoring = 'accuracy'
score = cross_val_score(RFC, X, y, cv=k_fold, n_jobs=1, scoring=scoring)
print(score)
round(np.mean(score)*100, 2)

[0.77419355 0.90322581 0.8        0.8        0.73333333 0.93333333
 0.66666667 0.8        0.73333333 0.86666667]


80.11

### 3.2.6 Naive Bayes Classifier

In [11]:
GNB = GaussianNB()
scoring = 'accuracy'
score = cross_val_score(GNB, X, y, cv=k_fold, n_jobs=1, scoring=scoring)
print(score)
round(np.mean(score)*100, 2)

[0.93548387 0.80645161 0.66666667 0.83333333 0.76666667 1.
 0.7        0.8        0.73333333 0.9       ]


81.42

### 3.2.7 SVM Classifier

In [12]:
SVC = SVC()
scoring = 'accuracy'
score = cross_val_score(SVC, X, y, cv=k_fold, n_jobs=1, scoring=scoring)
print(score)
round(np.mean(score)*100,2)

[0.70967742 0.67741935 0.53333333 0.5        0.56666667 0.8
 0.66666667 0.5        0.63333333 0.66666667]


62.54

### Results

Logistic Regression = 83.77

kNN Classifier = 64.88

Decision Tree Classifier = 76.15

Random Forset Classifier = 82.43

Naive Bayes Classifier = 81.42

SVM Classifier = 62.54

### Models from Higher to Worst

- Logistic Regression
- Random Forest Classifier
- Naive Bayes Classifier
- Decision Tree Classifier
- kNN Classifier
- SVM Classifier

So the best result was using Logistic Regression Classifier so we will test accuracy with it

**Conclusion:**

- High Blood Pressure, High Cholestrol and High Heart Rate leads to high chance of heart attack.

- In the count of target showed up that we have more chance of heart attack details.

- Age from 40-60 years have the high chance of heart attack.

- Male gender has more chance of heart attack compared to female ones.

- Highly Correlated factors in this dataset are :

    - Age and trtbps (blood pressure rate)
    - Age and chol (cholestrol level)